controllers/mysqlcluster_controller.go

```diff
@@ -260,18 +291,64 @@ func generateRandomBytes(n int) ([]byte, error) {
 	return ret, nil
 }
 
+func (r *MySQLClusterReconciler) createOrUpdateConfigMap(ctx context.Context, log logr.Logger, cluster *mocov1alpha1.MySQLCluster) error {
+	cm := &corev1.ConfigMap{}
+	cm.SetNamespace(cluster.Namespace)
+	cm.SetName(uniqueName(cluster))
+
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, cm, func() error {
+		cm.Labels = map[string]string{
+			appNameKey:      uniqueName(cluster),
+			appManagedByKey: myName,
+		}
+		gen := mysqlConfGenerator{
+			log: log,
+		}
+		gen.mergeSection("mysqld", defaultMycnf, false)
+		if cluster.Spec.MySQLConfigMapName != nil {
+			cm := &corev1.ConfigMap{}
+			err := r.Get(ctx, client.ObjectKey{Namespace: cluster.Namespace, Name: *cluster.Spec.MySQLConfigMapName}, cm)
+			if err != nil {
+				return err
+			}
+			gen.mergeSection("mysqld", cm.Data, false)
+		}
+		gen.merge(constMycnf, true)
+
+		myCnf, err := gen.generate()
+		if err != nil {
+			return err
+		}
+		cm.Data = make(map[string]string)
+		cm.Data[moco.MySQLConfName] = myCnf
+
+		return ctrl.SetControllerReference(cluster, cm, r.Scheme)
+	})
+	if err != nil {
+		log.Error(err, "unable to create-or-update ConfigMap")
+		return err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		log.Info("reconcile ConfigMap successfully", "op", op)
+	}
+	return nil
+}
+
 func (r *MySQLClusterReconciler) createOrUpdateHeadlessService(ctx context.Context, log logr.Logger, cluster *mocov1alpha1.MySQLCluster) error {
 	headless := &corev1.Service{}
 	headless.SetNamespace(cluster.Namespace)
-	headless.SetName(cluster.Name)
+	headless.SetName(uniqueName(cluster))
 
 	op, err := ctrl.CreateOrUpdate(ctx, r.Client, headless, func() error {
 		headless.Labels = map[string]string{
-			appNameKey: fmt.Sprintf("%s-%s", appName, cluster.Name),
+			appNameKey:      uniqueName(cluster),
+			appManagedByKey: myName,
 		}
```

Comment:
This pattern appears too often.  How about creating a utility function like this:

```go
func setLabels(om metav1.ObjectMeta) {
    om.Labels = map[string]string{
        appNameKey: om.Name,
        appManagedByKey: myName,
    }
}
```

ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1?tab=doc#ObjectMeta